primitives/Cargo.toml

```diff
@@ -18,7 +18,7 @@ exclude = ["tests", "contrib"]
 default = ["std"]
 std = ["alloc", "internals/std", "io/std", "units/std"]
 alloc = ["internals/alloc", "io/alloc", "units/alloc"]
-serde = ["dep:serde", "internals/serde", "units/serde"]
+serde = ["dep:serde", "internals/serde", "units/serde", "alloc"]
```

Comment:
We should try to avoid this before 1.0 because then it'll become kinda breaking. (Strictly, people should enable every feature they use directly but it's likely they'll miss it.) We should just bound those serde imlps on `all(feature = "serde", feature = alloc)`. Preferably in this PR but I can tolerate a separate PR if it's before release.